### DISCUSSION TITLE
CounterfactualUnitSelector Example

### DISCUSSION THREAD

**Discussion by aborundiya** [uber/causalml#346] [Q&A]
I was planing to use the CounterfactualUnitSelector to get the right audience for the upcoming campaign. Am bit lost as the CounterfactualUnitSelector has its own learner , how does the meta learner fits in this scheme ?

**Comment by aborundiya** [comment]
@t-tte @paullo0106 any thoughts on this ?

**Reply by t-tte** [reply]
Hi @aborundiya - the learner in the CounterfactualUnitSelector class is used to estimate the relevant conditional probabilities that Li and Pearl use to derive the bounds for the payoff from targeting a specific unit. No meta-learners are used in the class.

**Reply by aborundiya** [reply]
thanks @t-tte for the clarifications and was wondering why CounterfactualUnitSelector is in optimize library. I assumed that we may use the meta-learner to get the sensitive audience and use these audience as input to CounterfactualUnitSelector to get more refine audiences. What are your opinion on this ?

**Comment by t-tte** [comment]
If you want to use the output from eg a meta-learner to conduct value optimisation, you may want to look into the CounterfactualValueEstimator class: https://github.com/uber/causalml/blob/master/causalml/optimize/value_optimization.py Here is an example notebook using the approach: https://github.com/uber/causalml/blob/master/examples/counterfactual_value_optimization.ipynb
